src/python/grpcio_tests/tests/unit/server_ssl_cert_reload_test/_test.py

```diff
@@ -0,0 +1,364 @@
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This tests server certificate rotation support.
+
+This tests the ability of gRPC server to rotate its SSL cert for use
+in future channels with clients while not affecting any existing
+channel.
+
+For the credentials, we use two disjoint certificate hierarchies
+`cert_hier_1` and `cert_hier_2`.
+
+The test involves one server and two clients. Client 1 is a one-shot
+client: establish the secure channel, perform the RPC, and exit;
+client 2 establishes a secure channel, performs the RPC request, then
+sleeps for a specified amount, then performs a second RPC request on
+the same secure channel.
+
+All three start out using `cert_hier_1` and we check that everything
+works as expected, i.e., both client 1 and client 2 successfully
+perform RPCs.
+
+Then the server switches to `cert_hier_2`. Client 2, with its existing
+secure channel to the server, must be able to perform another RPC
+after the server's switch. Client 1 must fail while it continues to
+use `cert_hier_1`, and must succeed when it switches to `cert_hier_2`.
+
+All the "success" checks are based on the standard output/error of the
+two clients.
+"""
+
+import os
+import unittest
+from concurrent import futures
+
+import grpc
+from src.proto.grpc.testing import test_pb2_grpc
+
+from tests.interop import methods
```

Comment:
hmm actually regarding your "why put this test in `interop`" feedback, i guess i still have this here after having moved the test to `unit`. i'll try to address the other remaining issues first, and then i'll look into removing the `interop` stuff here if you say i should :) and if i should not use `interop` methods here, please let me know what to replace them with if you know off-hand. thanks!

Reply:
fixed with https://github.com/grpc/grpc/pull/13188/commits/a6cd8906fc5d0de133c42e36b8bf6905256386ff : no longer use `interop.modules`